lib/utils/import/simplenote/index.ts

```diff
@@ -47,31 +55,88 @@ class SimplenoteImporter extends EventEmitter {
         return;
       }
 
-      let dataObj;
-      try {
-        dataObj = JSON.parse(fileContent);
-      } catch (error) {
-        this.emit('status', 'error', 'Invalid json file.');
+      this.parseAndImportJson(fileContent);
+    };
+
+    fileReader.readAsText(file);
+  };
+
+  parseAndImportJson = (jsonContent) => {
+    const coreImporter = new CoreImporter(this.addNote);
+    let dataObj;
+
+    try {
+      dataObj = JSON.parse(jsonContent);
+    } catch (error) {
+      this.emit('status', 'error', 'Invalid JSON file.');
+      return;
+    }
+
+    if (!dataObj.activeNotes || !Array.isArray(dataObj.activeNotes)) {
+      this.emit('status', 'error', 'Invalid Simplenote JSON format.');
+      return;
+    }
+
+    const noteCount = dataObj.activeNotes.length + dataObj.trashedNotes.length;
+    const processedNotes = {
+      activeNotes: convertModificationDates(dataObj.activeNotes),
+      trashedNotes: convertModificationDates(dataObj.trashedNotes),
+    };
+
+    coreImporter.importNotes(processedNotes, this.options).then(() => {
+      this.emit('status', 'complete', noteCount);
+      this.recordEvent('importer_import_completed', {
+        source: 'simplenote',
+        note_count: noteCount,
+      });
+    });
+  };
+
+  processZipFile = (file) => {
+    const fileReader = new FileReader();
+
+    fileReader.onload = (event) => {
+      const fileContent = event.target.result;
+
+      if (!fileContent) {
+        this.emit('status', 'error', 'File was empty.');
         return;
       }
 
-      const noteCount =
-        dataObj.activeNotes.length + dataObj.trashedNotes.length;
-      const processedNotes = {
-        activeNotes: convertModificationDates(dataObj.activeNotes),
-        trashedNotes: convertModificationDates(dataObj.trashedNotes),
-      };
-
-      coreImporter.importNotes(processedNotes, this.options).then(() => {
-        this.emit('status', 'complete', noteCount);
-        this.recordEvent('importer_import_completed', {
-          source: 'simplenote',
-          note_count: noteCount,
+      import(/* webpackChunkName: 'jszip' */ 'jszip')
+        .then(({ default: JSZip }) => JSZip.loadAsync(fileContent))
+        .then((zip) => {
+          // Look for JSON files in the ZIP
+          const jsonFiles = Object.keys(zip.files).filter((filename) =>
+            filename.toLowerCase().endsWith('.json')
```

Comment:
we can look specifically for `source/notes.json` since that’s what the official exporter produces.

we don’t want ZIPs with random JSON destroying accounts with junk data